### PR TITLE
Adds support for Preonic rev 3

### PR DESCRIPTION
**Pull request by jackhumbert**
Also adds `matrix_col_t` which is dependent on the value of `MATRIX_ROWS` - I think this has been missing because not many boards try to take advantage of fancy row/col setups, or if they do, they increase the cols without making the rows go above 8.

### CODE REVIEW

**Comment by noroadsleft** [comment]
Curiosity's is getting the best of me:

https://github.com/qmk/qmk_firmware/blob/f34857c527e57428771b6a486d95f0ae430a6c94/keyboards/preonic/preonic.h#L8 vs. https://github.com/qmk/qmk_firmware/blob/f34857c527e57428771b6a486d95f0ae430a6c94/keyboards/preonic/preonic.h#L43

Any reason the matrix names can't be the same?

Not a complaint, just an observation.

**Comment by jackhumbert** [comment]
No reason! Just forgot to change that one over :) been trying to move everything towards "1x2u" since it's a better descriptor.

**Comment by noroadsleft** [comment]
Ha, I think I spotted something similar in the planck_rev6 branch a while back, and I was wondering if it was gonna stay like that.

File: tmk_core/common/matrix.h [hunk 1]
```diff
@@ -31,6 +31,16 @@ typedef  uint32_t   matrix_row_t;
 #error "MATRIX_COLS: invalid value"
 #endif
 
+#if (MATRIX_ROWS <= 8)
```
Comment (drashna) [inline tmk_core/common/matrix.h]: @That-Canadian I think this will fix the issue with the split boards I was having

**Pull request merged**